src/ErrorProne.NET/ErrorProne.NET.Test/ExceptionHandlingRules/ArgumentExceptionUsesStringAnalyzerTests.cs

```diff
@@ -0,0 +1,154 @@
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasNoParametersInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw [|new ArgumentException()|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasOnlyMessageParameterInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw [|new ArgumentException(""It's an argument exception"")|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasOnlyMessageParameterInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw [|new ArgumentException(""It's an argument exception"")|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasWrongParameterNameInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg1""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionMethodHasNoSuchParamName);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasWrongParameterNameInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg1""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionMethodHasNoSuchParamName);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasStringParameterNameInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameShouldNotBeString);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasStringParameterNameInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameShouldNotBeString);
+        }
+
+        [Test]
+        public void ShouldNotWarnOnArgumentExceptionHasNameofInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", nameof(arg));
```

Comment:
I think implementation is missing other cases, when for `paramName` was constructed by something else except string literal. I.e. this code is fine due to current implementation: `throw new ArgumentEception("foo", GetArgName())`. I'm not saying that this is not correct, but just curious what the behaviour should be in this case.